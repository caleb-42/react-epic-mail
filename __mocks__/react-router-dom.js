/* eslint-disable react/display-name */
/* eslint-disable react/prop-types */
import React from 'react';

const rrd = require('react-router-dom');

rrd.HashRouter = ({ children }) => <div>{children}</div>;


module.exports = rrd;