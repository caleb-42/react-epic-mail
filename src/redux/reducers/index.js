import { combineReducers } from 'redux';
import mails from './mailReducer';
import auth from './authReducer';
import activeNav from './navReducer';

const rootReducer = combineReducers({
  mails,
  auth,
  activeNav
});
export default rootReducer;