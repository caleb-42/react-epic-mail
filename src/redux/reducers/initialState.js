export const auth = {
  signUp: {},
  logIn: {}
}

export const mails = {
  messages: {
    data: [
      {
        id: 1,
        firstname: 'Sam',
        lastname: 'daniel',
        email: 'sam@gmail.com',
        subject: 'welcome home',
        message: "It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using 'Content here, content here', making it look like readable English. Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for 'lorem ipsum' will uncover many web sites still in their infancy. Various versions have evolved over the years, sometimes by accident,",
        senttime: '02-09-2019',
        status: 'read',
      },
      {
        id: 2,
        firstname: 'mathew',
        lastname: 'charles',
        email: 'mathew@gmail.com',
        message: 'Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source. Lorem Ipsum comes from sections 1.10.32 and 1.10.33 of "de Finibus Bonorum et Malorum" (The Extremes of Good and Evil) by Cicero, written in 45 BC. This book is a treatise on the theory of ethics, very popular during the Renaissance. The first line of Lorem Ipsum, "Lorem ipsum dolor sit amet..", comes from a line in section 1.10.32',
        subject: 'hi there',
        senttime: '02-09-2019',
        status: 'read',
      }
    ]
  },
  active: {

  }
};

export const navs = [
  {
    name: 'Mails',
    subNavs: [
      'Inbox',
      'Unread',
      'Sent',
      'Draft'
    ]
  },
  {
    name: 'Groups'
  },
  {
    name: 'Settings'
  },
]

export const activeNav = {
  menu: 'Mails', subMenu: 'Inbox'
}