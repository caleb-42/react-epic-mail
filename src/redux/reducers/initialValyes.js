export const signUp = {}

export const mails = []