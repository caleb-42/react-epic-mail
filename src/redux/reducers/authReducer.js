import * as types from '@redux/actions/actionTypes';
import { auth } from './initialState';

export default function authReducer(state = auth, action) {
  switch (action.type) {
    case types.SIGN_UP:
      return { ...state, signUp: action.signUp };
    case types.SETUP_USER:
      return {
        ...state,
        isAuthenticated: true,
        user: { ...action.payload },
      };
    case types.LOG_IN:
      return { ...state, logIn: action.logIn };
    default:
      return state;
  }
}