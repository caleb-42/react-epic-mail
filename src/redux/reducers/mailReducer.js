import * as types from '../actions/actionTypes';
import { mails } from './initialState';

export default function mailReducer(state = mails, action) {
  switch (action.type) {
    case types.CREATE_MAIL:
      return [...state, ...action.mail];
    default:
      return state;
  }
}