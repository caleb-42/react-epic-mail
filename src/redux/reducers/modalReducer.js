import * as types from '@redux/actions/actionTypes';
import { modal } from './initialState'

export default function activeNavReducer(state = modal, action) {
  switch (action.type) {
    case types.TOGGLE_MODAL:
      return { ...state, ...action.data }
    default:
      return { ...state }
  }
}