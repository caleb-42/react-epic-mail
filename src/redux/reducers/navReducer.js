import * as types from '@redux/actions/actionTypes';
import { activeNav } from './initialState'

export default function activeNavReducer(state = activeNav, action) {
  switch (action.type) {
    case types.NAVIGATE:
      return { ...action.activeNav, open: state.open }
    case types.TOGGLE_NAVIGATION:
      return { ...state, open: action.open }
    default:
      return { ...state }
  }
}