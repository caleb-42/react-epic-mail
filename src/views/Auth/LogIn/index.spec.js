import React from "react";
import renderer from "react-test-renderer";
import { cleanup, render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from 'react-router-dom';
import { Provider as ReduxProvider } from "react-redux";
import App from "@routes/app";
import { auth } from '@redux/reducers/initialState';
import configureStore from "@redux/configureStore";
import { LogIn } from './index'

const store = configureStore();

let defaultProps;

const renderLogInPage = (args) => {
  defaultProps = {
    history: { push: jest.fn((nav) => { }) },
    auth,
    logInUser: jest.fn((user, cb) => { cb({ data: true }); })
  }
  const props = { ...defaultProps, ...args };

  return render(
    <MemoryRouter>
      <LogIn {...props} />
    </MemoryRouter>
  );
}

describe('LOG IN', () => {

  afterEach(cleanup);

  describe('SNAPSHOT TEST', () => {
    it("log in page", () => {
      const tree = renderer.create(
        <ReduxProvider store={store}>
          <MemoryRouter initialEntries={['/signup']}>
            <App />
          </MemoryRouter>
        </ReduxProvider>
      );
      expect(tree).toMatchSnapshot();
    });
  });

  describe('COMPONENT TEST', () => {
    it("Should submit form if register button is clicked", () => {
      const { getByText } = renderLogInPage();
      fireEvent.click(getByText('Sign In'));
      expect(defaultProps.logInUser).toHaveBeenCalled();
    });
    it("Should navigate to home page if res.data is true", () => {
      const { getByText } = renderLogInPage({ auth: { ...auth, isAuthenticated: true } });
      /* fireEvent.click(getByText('Register')); */
      expect(defaultProps.history.push).toHaveBeenCalled();
    });
    it("Should not navigate to hpme page if res.data is false", () => {
      const { getByText } = renderLogInPage({
        logInUser: jest.fn((user, cb) => { cb({ data: false }); })
      });
      fireEvent.click(getByText('Sign In'));
    });
    it("Should call onChange function if form input is changed", () => {
      const { getByPlaceholderText } = renderLogInPage();
      const input = getByPlaceholderText('username@epicmail.com');
      fireEvent.change(input, { target: { value: '23' } });
    });
  });
});
