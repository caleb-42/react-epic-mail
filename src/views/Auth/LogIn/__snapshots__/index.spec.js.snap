// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LOG IN SNAPSHOT TEST log in page 1`] = `
<div>
  <nav
    className="lg:px-16 px-6 flex flex-wrap items-center lg:py-0 py-2 -mb-22 z-20 relative"
  >
    <div
      className="flex-1 flex justify-between items-center"
    >
      <a
        className="logo"
        href="#"
      />
    </div>
    <div
      className="hidden lg:flex lg:items-center lg:w-auto w-full"
      id="menu"
    >
      <div>
        <ul
          className="lg:flex items-center justify-between text-base text-white font-montserrat font-semibold pt-4 lg:pt-0"
        >
          <li>
            <a
              className="lg:p-4 py-3 px-0 block border-b-2 border-transparent hover:border-yellow-400"
              href="/signup"
              onClick={[Function]}
            >
              Features
            </a>
          </li>
          <li>
            <a
              className="lg:p-4 py-3 px-0 block border-b-2 border-transparent hover:border-yellow-400 lg:mb-0 mb-2"
              href="/signup"
              onClick={[Function]}
            >
              Support
            </a>
          </li>
          <li>
            <a
              className="lg:p-4 py-3 px-0 block border-b-2 border-transparent hover:border-yellow-400"
              href="/signup"
              onClick={[Function]}
            >
              Sign Up
            </a>
          </li>
          <li>
            <a
              className="lg:p-4 py-3 px-0 block border-b-2 border-transparent hover:border-yellow-400"
              href="/login"
              onClick={[Function]}
            >
              Log In
            </a>
          </li>
        </ul>
      </div>
      <a
        className="lg:ml-4 flex items-center justify-start lg:mb-0 mb-4 pointer-cursor"
        href="#"
      >
        <img
          alt="Andy Leverenz"
          className="rounded-full w-10 h-10 border-2 border-transparent hover:border-indigo-400"
          src="https://pbs.twimg.com/profile_images/1128143121475342337/e8tkhRaz_normal.jpg"
        />
      </a>
    </div>
  </nav>
  <div
    className="signUp flexdiv"
  >
    <form
      autoComplete="off"
      className="flexcenter inset-0 rounded-lg w-5/12 m-auto p-8"
      onSubmit={[Function]}
    >
      <h3
        className="text-center text-white text-3xl font-semibold"
      >
        Sign Up
      </h3>
      <div
        className="px-10 my-10"
      >
        <div
          className="flex input-group items-center"
        >
          <label
            htmlFor="email"
          >
            Email
          </label>
          <input
            className="input"
            name="email"
            onChange={[Function]}
            placeholder="username@epicmail.com"
            required={true}
            type="email"
          />
        </div>
        <div
          className="flex input-group items-center"
        >
          <label
            htmlFor="recoveryEmail"
          >
            Recovery Email
          </label>
          <input
            className="input"
            name="recoveryEmail"
            onChange={[Function]}
            required={true}
            type="email"
          />
        </div>
        <div
          className="flex input-group items-center"
        >
          <label
            htmlFor="password"
          >
            Password
          </label>
          <input
            className="input"
            name="password"
            onChange={[Function]}
            required={true}
            type="password"
          />
        </div>
        <div
          className="flex input-group items-center"
        >
          <label
            htmlFor="confirmPassword"
          >
            Confirm Password
          </label>
          <input
            className="input"
            name="confirmPassword"
            onChange={[Function]}
            required={true}
            type="password"
          />
        </div>
      </div>
      <p
        className="text-center block mx-auto text-white font-semibold my-2 opacity-75"
      >
        
      </p>
      <button
        className="px-8 py-3 shadow-lg text-center block mx-auto mt-4 text-yellow-800 font-semibold bg-yellow-500 rounded-lg"
        disabled={false}
        type="submit"
      >
        Register
      </button>
    </form>
  </div>
  <footer
    className="absolute bottom-0 text-center text-sm py-3 w-full"
  >
    <h3
      className="text-white font-lato"
    >
      EPICMAIL Copyright 
      2019
    </h3>
  </footer>
</div>
`;
